### PR TITLE
Add salvo telemetry

### PR DESCRIPTION
**Pull request by bmc-msft**
Adds telemetry fields for Salvo.

This PR should not be merged until approved by CELA.

Note: This PR includes #586

### CODE REVIEW

**Review by lzybkr — APPROVED** [review]
Changes look good.
Did you test the build with the optional features enabled?
Also note - the `z3-sys` crate build is quite slow, so we'll notice quickly if build artifact caching isn't working.

**Comment by bmc-msft** [comment]
> Changes look good.
> Did you test the build with the optional features enabled?

I built it locally.  I've not added it to CICD, though I'll do that now.

> Also note - the `z3-sys` crate build is quite slow, so we'll notice quickly if build artifact caching isn't working.

**Review by chkeita — APPROVED** [review]

**Comment by bmc-msft** [comment]
This PR is blocked by #671

File: src/agent/onefuzz-telemetry/src/lib.rs [hunk 1]
```diff
@@ -124,6 +147,37 @@ pub enum EventData {
...
+    UniqueCodeLocationCount(u64),
+    AverageInstructionsExecuted(f64),
+    MaxInstructionsExecuted(u64),
+    AverageTaintedInstructions(f64),
+    MaxTaintedInstructions(u64),
+    AverageMemoryTaintedInstructions(f64),
+    MaxMemoryTaintedInstructions(u64),
+    AveragePathLength(f64),
+    MaxPathLength(u64),
+    DiverganceRate(f64),
```
Comment (ranweiler) [inline src/agent/onefuzz-telemetry/src/lib.rs:168]: Throughout: s/Divergance/Divergence

File: src/agent/onefuzz-telemetry/src/lib.rs [hunk 2]
```diff
@@ -161,14 +215,53 @@ impl EventData {
...
+            Self::SymexTimeout(x) => ("symex_timeout", x.to_string()),
         }
     }
 
     pub fn can_share_with_microsoft(&self) -> bool {
         match self {
-            // TODO: Request CELA review of Version, as having this for central stats
-            //       would be useful to track uptake of new releases
-            Self::Version(_) => false,
+            Self::Version(_) => true,
```
Comment (ranweiler) [inline src/agent/onefuzz-telemetry/src/lib.rs:264]: 👍

File: src/agent/onefuzz-telemetry/src/lib.rs [hunk 3]
```diff
@@ -161,14 +215,53 @@ impl EventData {
             Self::ToolName(x) => ("tool_name", x.to_owned()),
             Self::Region(x) => ("region", x.to_owned()),
             Self::Role(x) => ("role", x.as_str().to_owned()),
+            #[cfg(feature = "intel_instructions")]
+            Self::MissedInstructionCode(x) => ("missed_instruction_code", format!("{:?}", x)),
```
Comment (ranweiler) [inline src/agent/onefuzz-telemetry/src/lib.rs:219]: Double-checking: we really want `:?`, here and below?

**Review by ranweiler — CHANGES_REQUESTED** [review]

File: src/agent/onefuzz-telemetry/src/lib.rs (hunk 3 above)
Comment (bmc-msft) [reply src/agent/onefuzz-telemetry/src/lib.rs:219]: Yes.  iced_x86::Code and iced_x86::Mnemonic do not implement Display.  However, they do implement fmt::Debug which provides what we want:

see iced_x86::Code:
https://github.com/icedland/iced/blob/315750d2ed73638c9bd5a687df6b5eeedd2fb1d1/src/rust/iced-x86/src/code.rs#L38859-L38864

See iced_x86::Mnemonic:
https://github.com/icedland/iced/blob/315750d2ed73638c9bd5a687df6b5eeedd2fb1d1/src/rust/iced-x86/src/mnemonic.rs#L3280-L3285

File: src/agent/onefuzz-telemetry/src/lib.rs (hunk 1 above)
Comment (ranweiler) [reply src/agent/onefuzz-telemetry/src/lib.rs:168]: Did you mean to push some changes?

Comment (bmc-msft) [reply src/agent/onefuzz-telemetry/src/lib.rs]: Thanks to my internet outage, I pushed from my laptop and then clicked resolve from my phone.

Thanks flaky internet!

**Review by ranweiler — APPROVED** [review]

**Pull request merged**